proposals/single-source-documentation.md

```diff
@@ -0,0 +1,79 @@
+# Single-source documentation 
+
+Authors: @johnnymatthews 
+
+Initial PR: TBD 
+
+## Purpose and impact 
+
+### Background and intent
+
+Readers can find all IPFS documentation on a single website or in a single GitHub repository.
+
+### Assumptions and hypotheses
+
+This project assumes that folks who want to learn about IPFS don't like hopping around blog posts, websites, and GitHub repos to find the information they need.
+
+### User workflow example
+
+1. The reader goes to docs.ipfs.io.
+1. They find the information need.
+1. They use that information to perform a task.
+    a. If the task failed, they return to docs.ipfs.io for more information.
+
+### Impact
+
+🔥🔥🔥
+
+Developers get upset when a project's documentation does not answer their questions. Having docs all over the place is an excellent way to make sure we don't answer user questions correctly. Therefore, having docs in one place is a straight-forward way to increase developers' chances of finding the information they need.
```

Comment:
A couple of other potential questions:
1. Will this work make it easier for contributors to make doc updates because they won't need to wonder where to make them?
2. Will our SEO improve at all with this?